pkg/transforms/event.go

```diff
@@ -0,0 +1,126 @@
+//
+// Copyright (c) 2022 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package transforms
+
+import (
+	"fmt"
+
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/util"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos/requests"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/common"
+	"github.com/edgexfoundry/go-mod-core-contracts/v2/dtos"
+)
+
+type EventWrapper struct {
+	profileName  string
+	deviceName   string
+	resourceName string
+	valueType    string
+	mediaType    string
+}
+
+// NewEventWrapperSimpleReading is provided to interact with EventWrapper to add a simple reading
+func NewEventWrapperSimpleReading(profileName string, deviceName string, resourceName string, valueType string) *EventWrapper {
+	eventWrapper := &EventWrapper{
+		profileName:  profileName,
+		deviceName:   deviceName,
+		resourceName: resourceName,
+		valueType:    valueType,
+	}
+	return eventWrapper
+}
+
+// NewEventWrapperBinaryReading is provided to interact with EventWrapper to add a binary reading
+func NewEventWrapperBinaryReading(profileName string, deviceName string, resourceName string, mediaType string) *EventWrapper {
+	eventWrapper := &EventWrapper{
+		profileName:  profileName,
+		deviceName:   deviceName,
+		resourceName: resourceName,
+		valueType:    common.ValueTypeBinary,
+		mediaType:    mediaType,
+	}
+	return eventWrapper
+}
+
+// NewEventWrapperObjectReading is provided to interact with EventWrapper to add a object reading type
+func NewEventWrapperObjectReading(profileName string, deviceName string, resourceName string) *EventWrapper {
+	eventWrapper := &EventWrapper{
+		profileName:  profileName,
+		deviceName:   deviceName,
+		resourceName: resourceName,
+		valueType:    common.ValueTypeObject,
+	}
+	return eventWrapper
+}
+
+// Wrap creates an EventRequest using the device name and reading name that have been set. If validation is turned on in
+// CoreServices then your deviceName and readingName must exist in the CoreMetadata and be properly registered in EdgeX.
```

Comment:
Event validation no long exists in 2.x
```suggestion
// Wrap creates an EventRequest using the Event/Reading metadata that have been set. 
```